akka-projection-jdbc/src/main/scala/akka/projection/jdbc/javadsl/JdbcProjection.scala

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright (C) 2020 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.projection.jdbc.javadsl
+
+import java.sql.Connection
+import java.util.concurrent.CompletableFuture
+import java.util.concurrent.CompletionStage
+import java.util.concurrent.ExecutorService
+import java.util.concurrent.Executors
+
+import scala.concurrent.ExecutionContext
+import scala.concurrent.Future
+import scala.util.control.NonFatal
+
+import akka.Done
+import akka.actor.typed.ActorSystem
+import akka.annotation.InternalApi
+import akka.projection.Projection
+import akka.projection.ProjectionId
+import akka.projection.javadsl.SourceProvider
+import akka.projection.javadsl.HandlerLifecycle
+import akka.japi.function.{ Function => JFunction }
+
+object JdbcProjection {
+
+  def exactlyOnce[Offset, Envelope, S <: JdbcSession](
+      projectionId: ProjectionId,
+      sourceProvider: SourceProvider[Offset, Envelope],
+      sessionFactory: () => S,
+      handler: JdbcHandler[Envelope, S]): JdbcProjection[Envelope] = ???
+
+}
+trait JdbcProjection[Envelope] extends Projection[Envelope] {
+
+  /**
+   * For testing purposes the offset table can be created programmatically.
+   * For production it's recommended to create the table with DDL statements
+   * before the system is started.
+   */
+  def createOffsetTableIfNotExists(system: ActorSystem[_]): CompletionStage[Done]
+}
+
+object JdbcSession {
+
+  /**
+   * INTERNAL API: run the blocking DB operations on a single thread one the passed EC.
+   */
+  @InternalApi
+  private[akka] def runBlockingDBIO[S <: JdbcSession, Result](jdbcSessionFactory: () => S)(block: S => Result)(
+      implicit ec: ExecutionContext): Future[Result] = {
+
+    // all blocking calls here
+    Future {
+      val session = jdbcSessionFactory()
+      try {
+        val result = block(session)
+        session.commit()
+        result
+      } catch {
+        case NonFatal(ex) =>
+          try {
+            session.rollback()
+          } catch {
+            case NonFatal(ex) => // the original exception is more interesting
+          }
+          throw ex
+      } finally {
+        session.close()
+      }
+    }
+  }
+}
+
+trait JdbcSession {
+  def withConnection[Result](func: JFunction[Connection, Result]): Result
+
+  def commit(): Unit
+  def rollback(): Unit
+  def close(): Unit
+}
+
+trait JdbcHandler[Envelope, S <: JdbcSession] extends HandlerLifecycle {
```

Comment:
`S <: JdbcSession` because we want the handler to receive the real instance from the user, not the JdbcSession trait. As such a user choose a JpaSession and access an `EntityManager`. 

We don't need (and don't want one) we only need the `Connection`.